docs/developer-docs/integrations/internet-identity/integrate-identity.md

```diff
@@ -37,18 +37,14 @@ DFINITY provides an [easy-to-use library (agent-js)](https://github.com/dfinity/
 
 These are the steps required to log in and use the obtained identity for canister calls:
 ```js
+import { AuthClient } from "@dfinity/auth-client";
+
 // First we have to create and AuthClient.
 const authClient = await AuthClient.create();
 
 // Call authClient.login(...) to login with Internet Identity. This will open a new tab
 // with the login prompt. The code has to wait for the login process to complete.
-// We can either use the callback functions directly or wrap in a promise.
-await new Promise((resolve, reject) => {
-  authClient.login({
-    onSuccess: resolve,
-    onError: reject,
-  });
-});
+await authClient.login();
```

Comment:
This won't actually work, even though login returns a promise 😅

The promise wrapper will actually work, but even then, I recommend that apps pass `onSuccess` and `onError` handlers, and avoid `await`ing `login`

Reply:
Okay I see what you're saying. Would it be feasible for this to work without a promise wrapper in v1.0? A precedent would be Auth0's SPA login flow (specifically [`loginWithRedirect()`](https://auth0.com/docs/libraries/auth0-single-page-app-sdk#use-promises)), which works exactly like this.

Reply:
Taking another look, it seems like the current promise-wrapping example causes a type error in TS (at least for some tsconfigs) caused by a mismatch between the input values for `onSuccess()` and `resolve(value)`. Just wanted to mention this since it's relevant to this conversation, although I'm sure most people will be able to figure out how to fix this.